lighthouse-core/report/html/renderer/psi.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* globals self DOM PerformanceCategoryRenderer Util DetailsRenderer */
+
+class PSI {
+  /**
+   * Returns all the elements that PSI needs to render the report
+   * We keep expose this helper method to minimize the 'public' API surface of the renderer
+   * and allow us to refactor without two-sided patches.
+   *
+   *   const {scoreGaugeEl, perfCategoryEl, finalScreenshotDataUri} = PSI.prepareLabData(
+   *      LHresponseJsonString,
+   *      document
+   *   );
+   *
+   * @param {string} LHresponseJsonString
```

Comment:
- can we call it an `LHRJsonString` or an `LHresultJsonString`? Or something else. Or do you think this will be the most self explanatory name?
- can always add param descriptions to help with that
- the capitalization scheme here is totally crazy. Capital `LH`, all lowercase `response`, camelcase `Json` :)
- can it just be an `LH.Result` and not a string? (i.e. already parsed)

Reply:
name updated to `LHResultJsonString`

> can it just be an LH.Result and not a string? (i.e. already parsed)

I considered that. The fact that it's a string is due to LR. (And me, obv). And at some point we'll fix LR and this won't be a string anymore and will be received on the PSI side as a proper `LH.Result` (or `LHLite.Result`) 

Since that seems like an implementation detail on the LH/LR side, it seemed reasonable that we eat the maintenance cost. Both LR and third_party/lh-renderer will have to roll in coordination, but that's about it.

Reply:
> Since that seems like an implementation detail on the LH/LR side, it seemed reasonable that we eat the maintenance cost. Both LR and third_party/lh-renderer will have to roll in coordination, but that's about it.

SG, just wondering